### DISCUSSION TITLE
Independent package focused on networking

### DISCUSSION THREAD

**Discussion by mohsenr** [zuhlke/Support#17] [Ideas]
Hi @zuhlke/iOS (and special guest @Milos 😁),

It’s been a few years and multiple projects since we started using the networking stack as defined in `Support`. We’ve add better abstractions (`HTTPService`, `HTTPEndpoint`) and we’ve switched to newer tech stacks a couple of times. (From `RxSwift`, to `Combine`, to `async`/`await`). At the same time, I feel like the core abstraction has been fairly stable over this time (`HTTPClient` and `HTTPRequest`).

So, I wanted to propose that we extract the networking portion of `Support` as an independent open source package. This allows us to better advertise the package as it will have a specific focus now. We should accompany this with better articles and documentation describing what the package does (Do check the documentation that we already have in the docc package).

At the same time, making this a separate package would also give us better focus on adding missing features (like, better built-in support for logging network requests, better support for hand-crafted and auto-recorded mock servers, contract testing, etc.).

To make sure I’m not falling for not invented here, I checked [Swift Package Index](https://swiftpackageindex.com) and I still find zero client packages with the fundamental design decision of our package: That the service definition and endpoint definition should be decoupled. So I do think that this is a good package for us to create.

Thoughts?

**Comment by Milos** [comment]
@mohsenr, thanks for the invite. Love the idea! 👍🚀

**Comment by mohsenr** [comment]
Let’s look into this in earnest in the new year.

In the meantime, the most difficult question of all: Package name suggestions are welcome 😁

**Reply by Milos** [reply]
How does "HTTPEndpoints" sound?

**Reply by ogezue** [reply]
Networking Extensions -> NetExt -> NeXT 😉

Steve would kill us.

**Reply by mohsenr** [reply]
😂

**Reply by Milos** [reply]
@mohsenr, is the enthusiasm still alive for this endeavour?
